sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/util/GcsUtil.java

```diff
@@ -208,6 +211,8 @@ private GcsUtil(
     this.httpRequestInitializer = httpRequestInitializer;
     this.uploadBufferSizeBytes = uploadBufferSizeBytes;
     this.executorService = executorService;
+    this.batchExecutorService =
```

Comment:
Instead of storing this here, why not just drop the "exiting" part of the executor service below?

This way the thread pool has a limited lifecycle (when its being used) otherwise it gets dropped and the behavior will be very similar to before.

Reply:
Yes, that would have the least chance of introducing new issues.